File: ecobertura.ui/src/ecobertura/ui/editors/WindowListener.java
package ecobertura.ui.editors;

import org.eclipse.ui.IPartListener2;
import org.eclipse.ui.IWindowListener;
import org.eclipse.ui.IWorkbenchPartReference;
import org.eclipse.ui.IWorkbenchWindow;

final class WindowListener implements IWindowListener {
	private EditorTracker editorTracker;
	
	WindowListener(EditorTracker editorTracker) {
		this.editorTracker = editorTracker;
	}
	
	private final IPartListener2 partListener = new IPartListener2() {
		public void partOpened(IWorkbenchPartReference partref) {
			editorTracker.annotateEditor(partref);
		}

		public void partActivated(IWorkbenchPartReference partref) {}
		public void partBroughtToTop(IWorkbenchPartReference partref) {}
		public void partVisible(IWorkbenchPartReference partref) {}
		public void partInputChanged(IWorkbenchPartReference partref) {}
		public void partClosed(IWorkbenchPartReference partref) {}
		public void partDeactivated(IWorkbenchPartReference partref) {}
		public void partHidden(IWorkbenchPartReference partref) {}
	};
	
	public void windowOpened(IWorkbenchWindow window) {
		window.getPartService().addPartListener(partListener);
	}

	public void windowClosed(IWorkbenchWindow window) {
		window.getPartService().removePartListener(partListener);
	}

	public void windowActivated(IWorkbenchWindow window) { /* NOP */ }
	public void windowDeactivated(IWorkbenchWindow window) { /* NOP */ }

	IPartListener2 getPartListener() {
		return partListener;
	}
}

File: ecobertura.ui/src/main/scala/ecobertura/ui/editors/WindowListener.scala
/*
 * This file is part of eCobertura.
 * 
 * Copyright (c) 2009, 2010 Joachim Hofer
 * All rights reserved.
 *
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 */
package ecobertura.ui.editors

import org.eclipse.ui._

class WindowListener(editorTracker: EditorsAnnotator) extends IWindowListener {

	val partListener: IPartListener2 = new IPartListener2() {
		override def partOpened(partref: IWorkbenchPartReference) = 
			editorTracker.annotateEditor(partref)
		
		override def partActivated(partref: IWorkbenchPartReference) = { /* NOP */ } 
		override def partBroughtToTop(partref: IWorkbenchPartReference) = { /* NOP */ } 
		override def partVisible(partref: IWorkbenchPartReference) = { /* NOP */ } 
		override def partInputChanged(partref: IWorkbenchPartReference) = { /* NOP */ } 
		override def partClosed(partref: IWorkbenchPartReference) = { /* NOP */ } 
		override def partDeactivated(partref: IWorkbenchPartReference) = { /* NOP */ } 
		override def partHidden(partref: IWorkbenchPartReference) = { /* NOP */ } 
	}
	
	override def windowOpened(window: IWorkbenchWindow) =
		window.getPartService.addPartListener(partListener)
	
	override def windowClosed(window: IWorkbenchWindow) =
		window.getPartService.removePartListener(partListener)
		
	override def windowActivated(window: IWorkbenchWindow) = { /* NOP */ }
	override def windowDeactivated(window: IWorkbenchWindow) = { /* NOP */ }
}
